### PR TITLE
feat(cb2-7045): create from search

### PR DESCRIPTION
**Pull request by JunYanBJSS**
## feat(cb2-7045): create a new Tech Record when a search returns no results

Handle the case where no results return when doing a search
[CB2-7045](https://dvsa.atlassian.net/browse/CB2-7045)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

**Review by kcarr-bjss — APPROVED** [review]

**Review by shivangidas — APPROVED** [review]

File: src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts [hunk 1]
```diff
@@ -69,6 +83,24 @@ describe('MultipleSearchResultsComponent', () => {
...
+    beforeEach(() => {
+      store = TestBed.inject(MockStore);
+      techRecordHttpService = TestBed.inject(TechnicalRecordHttpService);
+      store.overrideSelector(selectQueryParams, { vin: '123456' });
+
+      fixture = TestBed.createComponent(MultipleSearchResultsComponent);
+      component = fixture.componentInstance;
+
+      fixture.detectChanges();
+    });
```
Comment (gjulien-bjss) [inline src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:96]: do you need to re-create the component?

Comment (shivangidas) [reply src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:96]: this test may not need a new describe either. Could  the 'default tests' describe have the two 'it' conditions - when there are search results and when there are none

Comment (JunYanBJSS) [reply src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:96]: > do you need to re-create the component?, not convinced you need the beforeEach here

Yes mate. I will get rid of those lines

**Review by gjulien-bjss — DISMISSED** [review]

File: src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts [hunk 2]
```diff
@@ -56,8 +58,25 @@ describe('MultipleSearchResultsComponent', () => {
...
     });
 
-    it('should navigate back', fakeAsync(() => {
+    it('should return not found error and create link when searchResults is null', async () => {
+      const button = fixture.debugElement.query(By.css('.govuk-link'));
+      expect(button).toBeTruthy();
+    });
+
+    it('should navigate back when searchResults not null', fakeAsync(() => {
       const navigateBackSpy = jest.spyOn(component, 'navigateBack');
```
Comment (owen-corrigan-bjss) [inline src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:67]: Do you need this navigateBackSpy? it doesn't appear to be being used

File: src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts [hunk 3]
```diff
@@ -56,8 +58,25 @@ describe('MultipleSearchResultsComponent', () => {
       expect(component).toBeTruthy();
     });
 
-    it('should navigate back', fakeAsync(() => {
+    it('should return not found error and create link when searchResults is null', async () => {
```
Comment (shivangidas) [inline src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:61]:
```suggestion
    it('should show create link when searchResults is null', async () => {
```

**Review by shivangidas — APPROVED** [review]

File: src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts (hunk 2 above)
Comment (JunYanBJSS) [reply src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts:67]: I think we need it. We expect navigateBackSpy function could be called whenever back link is clicked.

File: src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts (hunk 3 above)
Comment (JunYanBJSS) [reply src/app/features/search/multiple-search-results/multiple-search-results.component.spec.ts]: have amended now

**Review by shivangidas — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Pull request merged**